### PR TITLE
DEVPROD-976 Let users specify an owner and repo for modules

### PR DESCRIPTION
**Pull request by malikchaya2**
DEVPROD-976

### Description
Right now, we expect users to specify a git url for modules. This is the example we show in our docs:

```
modules:
  name: example
  repo: git@github.com:10gen/mongo-example-modules.git
  prefix: src/mongo/db/modules
  ref: 12341a65256ff78b6d15ab79a1c7088443b9abcd
  branch: master
```

Now that we are deprecating ssh cloning, users should switch over to this:

```
modules:
  name: example
  owner: 10gen
  repo: mongo-example-modules
  prefix: src/mongo/db/modules
  ref: 12341a65256ff78b6d15ab79a1c7088443b9abcd
  branch: master
```

so that we can build the clone url ourselves.

### Testing
Updated tests 

### Documentation
Doc updates included

### CODE REVIEW

File: agent/command/git.go [hunk 1]
```diff
@@ -719,32 +719,34 @@ func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
...
-		repo:     repo,
 		branch:   "",
 		dir:      moduleBase,
 		method:   cloneMethod,
+		location: module.Repo,
 	}
+
 	// If the module repo is using the deprecated ssh cloning method, extract the owner
-	// and repo from the string and construct a https cloning link manually.
+	// and repo from the string and save it to ops so that the an https cloning link
```
Comment (hadjri) [inline agent/command/git.go:731]: nit: ops --> clone options or cloneOpts

File: agent/command/git.go [hunk 2]
```diff
@@ -719,32 +719,34 @@ func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
...
 	// This is a temporary workaround which will be removed once users have switched over.
-	if strings.Contains(opts.location, "git@github.com:") {
+	owner, repo, err := module.GetOwnerAndRepo()
+	if err != nil {
+		return errors.Wrapf(err, "getting module owner and repo '%s'", module.Name)
+	}
+
+	opts.owner = owner
+	opts.repo = repo
+	if strings.Contains(module.Repo, "git@github.com:") {
```
Comment (hadjri) [inline agent/command/git.go:741]: Hmm, maybe we should change that L742 to a warning so it stands out more? Perhaps we can either mention to go to the docs, or provide a link to the docs you've written here as well

**Review by hadjri — CHANGES_REQUESTED** [review]

File: agent/command/git.go (hunk 2 above)
Comment (malikchaya2) [reply agent/command/git.go:741]: Changed to a warning. I don't know if it's a good idea to link to docs in task logs though, I don't think we have a precedent for that. I'm hoping that users who are confused will know to look for docs or reach out in users.

File: agent/command/git.go (hunk 1 above)
Comment (malikchaya2) [reply agent/command/git.go]: done

File: model/project.go [hunk 3]
```diff
@@ -456,11 +456,19 @@ type Module struct {
 	Name       string `yaml:"name,omitempty" bson:"name" plugin:"expand"`
 	Branch     string `yaml:"branch,omitempty" bson:"branch"  plugin:"expand"`
 	Repo       string `yaml:"repo,omitempty" bson:"repo"  plugin:"expand"`
+	Owner      string `yaml:"owner,omitempty" bson:"owner"  plugin:"expand"`
 	Prefix     string `yaml:"prefix,omitempty" bson:"prefix"  plugin:"expand"`
 	Ref        string `yaml:"ref,omitempty" bson:"ref"  plugin:"expand"`
 	AutoUpdate bool   `yaml:"auto_update,omitempty" bson:"auto_update"`
 }
 
+func (m Module) GetOwnerAndRepo() (string, string, error) {
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/project.go:468]: done

File: operations/fetch.go [hunk 4]
```diff
@@ -305,9 +305,10 @@ func cloneSource(task *service.RestTask, project *model.ProjectRef, config *mode
 
 		moduleBase := filepath.Join(cloneDir, modulePrefix, module.Name)
 		fmt.Printf("Fetching module %v at %v\n", moduleName, module.Branch)
-		owner, repo, err := thirdparty.ParseGitUrl(module.Repo)
+
+		owner, repo, err := module.GetOwnerAndRepo()
 		if err != nil {
-			return errors.Wrapf(err, "parsing git URL '%s'", module.Repo)
+			return errors.Wrapf(err, "error getting owner and repo for '%s'", module.Name)
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply operations/fetch.go]: done

File: agent/command/git.go (hunk 2 above)
Comment (hadjri) [reply agent/command/git.go:741]: Fair enough, SGTM

**Review by hadjri — APPROVED** [review]

**Pull request merged**